test/unit/api_spec.js

```diff
@@ -338,25 +358,31 @@ describe('api', function() {
     it('gets page labels', function () {
       // PageLabels with Roman/Arabic numerals.
       var url0 = combineUrl(window.location.href, '../pdfs/bug793632.pdf');
-      var promise0 = PDFJS.getDocument(url0).promise.then(function (pdfDoc) {
-        return pdfDoc.getPageLabels();
-      });
+      var loadingTask0 = PDFJS.getDocument(url0);
+
       // PageLabels with only a label prefix.
       var url1 = combineUrl(window.location.href, '../pdfs/issue1453.pdf');
-      var promise1 = PDFJS.getDocument(url1).promise.then(function (pdfDoc) {
-        return pdfDoc.getPageLabels();
-      });
+      var loadingTask1 = PDFJS.getDocument(url1);
+
       // PageLabels identical to standard page numbering.
       var url2 = combineUrl(window.location.href, '../pdfs/rotation.pdf');
-      var promise2 = PDFJS.getDocument(url2).promise.then(function (pdfDoc) {
-        return pdfDoc.getPageLabels();
-      });
+      var loadingTask2 = PDFJS.getDocument(url2);
 
-      waitsForPromiseResolved(Promise.all([promise0, promise1, promise2]),
-          function (pageLabels) {
-        expect(pageLabels[0]).toEqual(['i', 'ii', 'iii', '1']);
-        expect(pageLabels[1]).toEqual(['Front Page1']);
-        expect(pageLabels[2]).toEqual([]);
+      var promises = [loadingTask0.promise, loadingTask1.promise,
+                      loadingTask2.promise];
+      Promise.all(promises).then(function (documents) {
```

Comment:
Doesn't removing `waitsForPromiseResolved` cause issues if one (or more) of the promises is rejected, since then the test will pass despite not actually testing anything?

You probably need to do something like this instead, to ensure that you catch every possible failure mode:
```diff
@@ -338,17 +338,20 @@ describe('api', function() {
     it('gets page labels', function () {
       // PageLabels with Roman/Arabic numerals.
       var url0 = combineUrl(window.location.href, '../pdfs/bug793632.pdf');
-      var promise0 = PDFJS.getDocument(url0).promise.then(function (pdfDoc) {
+      var loadingTask0 = PDFJS.getDocument(url0);
+      var promise0 = loadingTask0.promise.then(function (pdfDoc) {
         return pdfDoc.getPageLabels();
       });
       // PageLabels with only a label prefix.
       var url1 = combineUrl(window.location.href, '../pdfs/issue1453.pdf');
-      var promise1 = PDFJS.getDocument(url1).promise.then(function (pdfDoc) {
+      var loadingTask1 = PDFJS.getDocument(url1);
+      var promise1 = loadingTask1.promise.then(function (pdfDoc) {
         return pdfDoc.getPageLabels();
       });
       // PageLabels identical to standard page numbering.
       var url2 = combineUrl(window.location.href, '../pdfs/rotation.pdf');
-      var promise2 = PDFJS.getDocument(url2).promise.then(function (pdfDoc) {
+      var loadingTask2 = PDFJS.getDocument(url2);
+      var promise2 = loadingTask2.promise.then(function (pdfDoc) {
         return pdfDoc.getPageLabels();
       });

@@ -357,6 +360,10 @@ describe('api', function() {
         expect(pageLabels[0]).toEqual(['i', 'ii', 'iii', '1']);
         expect(pageLabels[1]).toEqual(['Front Page1']);
         expect(pageLabels[2]).toEqual([]);
+
+        loadingTask0.destroy();
+        loadingTask1.destroy();
+        loadingTask2.destroy();
       });
     });
     it('gets attachments', function() {
```

*Note that this might be an issue elsewhere in the patch too, I've not looked through all of it.*